packages/typespec-azure-core/src/decorators.ts

```diff
@@ -268,15 +267,11 @@ export function getItems(program: Program, entity: Type): boolean | undefined {
   return program.stateMap(AzureCoreStateKeys.items).get(entity);
 }
 
-export const $nextLink: NextLinkDecorator = (context: DecoratorContext, entity: ModelProperty) => {
-  context.program.stateMap(AzureCoreStateKeys.nextLink).set(entity, true);
-};
-
 /**
  * Returns `true` if the property is marked with `@nextLink`.
  */
-export function getNextLink(program: Program, entity: Type): boolean | undefined {
-  return program.stateMap(AzureCoreStateKeys.nextLink).get(entity);
+export function getNextLink(program: Program, entity: ModelProperty): boolean {
+  return program.stateSet(Symbol.for(`TypeSpec.nextLink`)).has(entity);
```

Comment:
Instead, can we call the accessor function in typespec core?

Reply:
I tried that, but I got this error: https://github.com/Azure/azure-rest-api-specs-pr/actions/runs/12146646232/job/33871047296#step:5:29355 
![image](https://github.com/user-attachments/assets/caeb9ff1-6ed5-47af-ac53-17650fcd1f9f) I called this one: https://github.com/microsoft/typespec/blob/968312e3d0d456b74981bfe244a1ee0d01a9beda/packages/compiler/src/lib/decorators.ts#L124

is there another one?

Reply:
There is an `isNextLink` function, but it isn't exported.  I think we can stay with the current implementation, but mark the accessor as deprecated

Reply:
thanks Mark! that change needs to be done in a different pull request in typespec repository, right?